ndn_hydra/repo/handles/query_handle.py

```diff
@@ -57,9 +57,14 @@ def unlisten(self, prefix):
 
     def _on_interest(self, int_name, int_param, _app_param):
         if not int_param.must_be_fresh or not int_param.can_be_prefix:
+            return 
+        if not _app_param:
+            self.logger.info('Query handle: No querytype')
             return
+
         query = self._get_query_from_interest(Name.to_str(int_name))
```

Comment:
Since the query is the AppParams, you can say:
query = _app_params

This would let you get rid of the _get_query_from_interest function completely. (delete it)

Reply:
Gotcha. Removed that code in most recent commit.